### PR TITLE
Split out WMR hand definition and WMR hand mesh provider

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview

To reduce the amount of platform-specific code stored in a strict inheritance chain, I've split out the mesh code from the controller definition concept. Now, WMR articulated hands can use the base definition, while still defining their own way of providing the hand mesh (in this case, directly from the Windows APIs. In the future, other hand definitions may grab this from XR plug-in extensions).

### CODE REVIEW

File: Assets/MRTK/Providers/WindowsMixedReality/Shared/WindowsMixedRealityHandMeshProvider.cs [hunk 1]
```diff
@@ -3,24 +3,28 @@
...
 #endif // WINDOWS_UWP
 
 namespace Microsoft.MixedReality.Toolkit.WindowsMixedReality
 {
     /// <summary>
-    /// Defines the additional data, like hand mesh, that an articulated hand on HoloLens 2 can provide.
+    /// Queries the hand mesh data that an articulated hand on HoloLens 2 can provide.
     /// </summary>
-    public class WindowsMixedRealityArticulatedHandDefinition : ArticulatedHandDefinition
+    public class WindowsMixedRealityHandMeshProvider
```
Comment (CDiaz-MS) [inline Assets/MRTK/Providers/WindowsMixedReality/Shared/WindowsMixedRealityHandMeshProvider.cs:21]: Is this considered a breaking change and if it is should we add it to the release notes?

Comment (keveleigh) [reply Assets/MRTK/Providers/WindowsMixedReality/Shared/WindowsMixedRealityHandMeshProvider.cs:21]: Ah, yeah the class no longer existing would be a breaking change, though it shouldn't really have any use outside of these classes. I'll re-add and obsolete.

Comment (keveleigh) [reply Assets/MRTK/Providers/WindowsMixedReality/Shared/WindowsMixedRealityHandMeshProvider.cs]: Re-added!

**Pull request merged**